CMakeLists.txt

```diff
@@ -549,6 +550,15 @@ target_include_directories(openPMD PUBLIC
     $<INSTALL_INTERFACE:include>
 )
 
+# Winsock library for gethostname() function on Windows
+if(WIN32)
+    # need to link this publically as WSAStartup() and WSACleanup() need to be
+    # called surrounding the gethostname() function on Windows
+    # and it needs to be done at client site since the winsocks API is
+    # initialized statically per process....
+    target_link_libraries(openPMD PUBLIC ws2_32)
```

Comment:
Would it be more portable (MSYS/MinGW, MSVC, LLVM on Windows) if this was searched for via [find_library](https://cmake.org/cmake/help/latest/command/find_library.html)?

Reply:
Probably yes, will add this tomorrow when I get my hands on a Windows computer again.